conda_smithy/configure_feedstock.py

```diff
@@ -1594,6 +1595,61 @@ def make_jinja_env(feedstock_directory):
     return env
 
 
+def get_migrations_in_dir(migrations_root):
+    res = {}
+    for fn in glob.glob(os.path.join(migrations_root, "*.yaml")):
+        with open(fn, "r") as f:
+            contents = f.read()
+            migration_yaml = (
+                yaml.load(contents, Loader=yaml.loader.BaseLoader) or {}
+            )
+            # Use a object as timestamp to not delete it
+            ts = migration_yaml.get("migrator_ts", object())
+            migration_number = migration_yaml.get("__migrator", {}).get(
+                "migration_number", 1
+            )
+            res[ts] = (fn, migration_number)
+    return res
+
+
+def set_migration_fns(forge_dir, forge_config):
+    exclusive_config_file = forge_config["exclusive_config_file"]
+    cfp_migrations_dir = os.path.join(
+        os.path.dirname(exclusive_config_file),
+        "share",
+        "conda-forge",
+        "migrations",
+    )
+
+    migrations_root = os.path.join(forge_dir, ".ci_support", "migrations")
+    migrations_in_feedstock = get_migrations_in_dir(migrations_root)
+
+    if not os.path.exists(cfp_migrations_dir):
```

Comment:
@beckermr, I have this condition here to not delete stuff if the migrations in cfp don't exist.

Reply:
I’m confused. What does this snippet of text from the pr description mean?

“ remove if the migration yaml has a
timestamp and there's no corresponding one in conda-forge-pinning
which indicates that the migration is over.”

Reply:
Like how do we distinguish between “no migrations happening” and “we are using the old system”?

It seems like we need to throw the switch once.

Reply:
I first check that `$PREFIX/share/conda-forge/migrations` exist. If not, go back to old system.
If it does, I check in `<feedstock-root>/.ci_support/migrations` and compare with `$PREFIX/share/conda-forge/migrations`. If a migration exist in `<feedstock-root>` and doesn't on `$PREFIX`, it's deleted.